FEXCore/Source/Interface/IR/Passes/RegisterAllocationPass.cpp

```diff
@@ -265,37 +269,42 @@ class ConstrainedRAPass final : public RegisterAllocationPass {
     }
   };
 
+#define foreach_bit(b, x) for (uint32_t __x = (x), b; ((b) = __builtin_ffs(__x) - 1, __x); __x &= ~(1 << (b)))
```

Comment:
This should probably be commented about the magic bit-walking that this uses. I know what it is doing but the magic may be mysterious for anyone that isn't used to walking discontiguous bits in a mask.

I like it.

Reply:
done.